tool/tctl/common/resource_command.go

```diff
@@ -551,40 +552,37 @@ func (rc *ResourceCommand) IsForced() bool {
 }
 
 // getCollection lists all resources of a given type
-func (rc *ResourceCommand) getCollection(client auth.ClientI) (c ResourceCollection, err error) {
-	ctx := context.TODO()
+func (rc *ResourceCommand) getCollection(client auth.ClientI) (ResourceCollection, error) {
 	if rc.ref.Kind == "" {
 		return nil, trace.BadParameter("specify resource to list, e.g. 'tctl get roles'")
 	}
+
+	// TODO: pass the context from CLI to terminate requests on Ctrl-C
+	ctx := context.TODO()
 	switch rc.ref.Kind {
-	// load user(s)
 	case services.KindUser:
-		var users services.Users
-		// just one?
-		if !rc.ref.IsEmpty() {
-			user, err := client.GetUser(rc.ref.Name, rc.withSecrets)
-			if err != nil {
-				return nil, trace.Wrap(err)
-			}
-			users = services.Users{user}
-			// all of them?
-		} else {
-			users, err = client.GetUsers(rc.withSecrets)
+		if rc.ref.Name == "" {
+			users, err := client.GetUsers(rc.withSecrets)
 			if err != nil {
 				return nil, trace.Wrap(err)
 			}
+			return &userCollection{users: users}, nil
 		}
-		return &userCollection{users: users}, nil
+		user, err := client.GetUser(rc.ref.Name, rc.withSecrets)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		return &userCollection{users: services.Users{user}}, nil
 	case services.KindConnectors:
-		sc, err := client.GetSAMLConnectors(ctx, rc.withSecrets)
+		sc, err := getSAMLConnectors(ctx, client, rc.ref.Name, rc.withSecrets)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		oc, err := client.GetOIDCConnectors(ctx, rc.withSecrets)
+		oc, err := getOIDCConnectors(ctx, client, rc.ref.Name, rc.withSecrets)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		gc, err := client.GetGithubConnectors(ctx, rc.withSecrets)
+		gc, err := getGithubConnectors(ctx, client, rc.ref.Name, rc.withSecrets)
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
```

Comment:
Handle `trace.IsNotFound(err)` for each case.
And return an error if all results are empty and `rc.ref.Name != ""`

For example, a user might try `tctl get connectors/my-github-connector` which would fail the fetch on saml/oidc connectors. Similarly, `tctl get connectors/nonexistent-connector` should fail.